core/benches/bench_cell.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use criterion::{
+    black_box,
+    criterion_group,
+    criterion_main,
+    BenchmarkId,
+    Criterion,
+};
+use ink_core::{
+    env,
+    storage2::{
+        lazy::LazyCell,
+        traits::{
+            KeyPtr,
+            SpreadLayout,
+        },
+    },
+};
+use ink_primitives::Key;
+
+criterion_group!(populated_cache, bench_set_populated_cache);
+criterion_group!(empty_cache, bench_set_empty_cache,);
+criterion_main!(populated_cache, empty_cache);
+
+/// For the populated cache benchmarks we use a loop of 100 ops, since
+/// for only one op the criterion benchmark might just show 0 ps and thus
+/// make it hard to estimate the difference between `deref_mut` and `set`.
+mod populated_cache {
+    use super::*;
+
+    pub fn set() {
+        let mut cell = <LazyCell<i32>>::new(Some(1));
+        for i in 0..100 {
+            black_box(cell.set(i));
+        }
```

Comment:
I am not sure the loops here hurt the actual measurement. Are they needed? If yes, why?

Reply:
The comment above the `mod` describes the reason:
```
/// For the populated cache benchmarks we use a loop of 100 ops, since
/// for only one op the criterion benchmark might just show 0 ps and thus
/// make it hard to estimate the difference between `deref_mut` and `set`.
```
I agree with you though, but if I remove the loop criterion always panics because of too few meaningful samples. I investigated this in more detail for a while yesterday and don't yet fully understand the issue. I'll look more into it next week, but from what I understand so far it could also be a bug/issue with criterion (or of course the benchmark is wrong).